Expensify/App
# Notifications redirection fix

When a mobile notification from User A is clicked, Chat isn't opening to the chat with User A

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/144636

### Tests (iOS and/or Android)
- You need to run this in your personal device, this [SO](https://stackoverflow.com/c/expensify/questions/6735) helps.
- You also need 3 users for this test, one to test on the device User1, another to be UserA and another to be UserB. I just added the other 2 users to User1 policy.

1. run the app on your device and login with User1, accept notifications.
1. run the app on chrome, and login with UserA
1. run the app on firefox or safary and login with UserB
1. Send a message to User1 from UserA, verify that it arrives and you get a push, and when you tap it you go to that conversation.
1. Send a message to User1 from UserB, verify that it arrives and you get a push, and when you tap it you go to that conversation.
1. Kill the app and restart it, verify that you don't have to login again.
1. Send a message to User1 from UserA, verify that it arrives and you get a push, and when you tap it you go to that conversation.
1. Send a message to User1 from UserB, verify that it arrives and you get a push, and when you tap it you go to that conversation.
1. Put the app into background.
1. Send a message to User1 from UserA, verify that it arrives and you get a push. Verify that when you tap it the app comes into the foreground and you are directed to the conversation with UserA.
